### PR TITLE
Add run.sh script to generate all API artifacts.

### PR DESCRIPTION
**Pull request by mikelehen**

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+set -o nounset
+set -o errexit
+
+DEFAULT_DATA_SOURCES_DIR="../covid-data-public"
+DEFAULT_API_RESULTS_DIR="api_results"
+
+main () {
+  # Go to repo root (where run.sh lives).
+  cd "$(dirname "$0")"
```
Comment (ghop02) [inline run.sh:14]: you could do this using `pushd` and `popd` so that it doesn't change your directory after the run

File: run.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  # TODO: Move DoD onto a more general-purpose schema rather than treat them custom.
+  DOD_DIR="${API_OUTPUT_DIR}/custom1"
+
+  echo ">>> Generating state models to ${STATES_DIR}"
+  # TODO(#148): We need to clean up the output of these scripts!
+  ./run_model.py state -o "${API_OUTPUT_DIR}" > /dev/null
+
+  echo ">>> Generating county models to ${COUNTIES_DIR}"
+  # TODO(#148): We need to clean up the output of these scripts!
+  ./run_model.py county -o "${COUNTIES_DIR}" > /dev/null
```
Comment (ghop02) [inline run.sh:64]: by running this inside a function, i don't think it'll automatically fail on a non zero response.  
a) is this true?
b) if so we should probably check the return value and exit if necessary

File: run.sh (hunk 1 above)
Comment (mikelehen) [reply run.sh:14]: `cd` within a bash script doesn't affect the current directory of the caller.  So `pushd` / `popd` isn't necessary.

File: run.sh (hunk 2 above)
Comment (mikelehen) [reply run.sh:64]: a) not true AFAIK!

My understanding is that the `set -o errexit` at the top will automatically fail the entire script if any command fails.  Feel free to prove me wrong. 😛

File: run.sh (hunk 1 above)
Comment (ghop02) [reply run.sh:14]: time to go brush up on my bash 😀

File: run.sh (hunk 2 above)
Comment (ghop02) [reply run.sh:64]: you are right!

File: .gitignore [hunk 3]
```diff
@@ -1,5 +1,6 @@
 .~lock.*
 results/*
+api_results/
```
Comment (fridiculous) [inline .gitignore:3]:
```suggestion
api_results/*
```

File: run.sh [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  # requirement that the county model JSON is in a /county subdirectory of the
+  # states?
+  COUNTIES_DIR="${API_OUTPUT_DIR}/county";
+  COUNTY_SUMMARIES_DIR="${API_OUTPUT_DIR}/county_summaries";
+  CASE_SUMMARIES_DIR="${API_OUTPUT_DIR}/case_summary"
+
+  # TODO: Move DoD onto a more general-purpose schema rather than treat them custom.
+  DOD_DIR="${API_OUTPUT_DIR}/custom1"
+
+  echo ">>> Generating state models to ${STATES_DIR}"
```
Comment (fridiculous) [inline run.sh:58]: it seems like up to this point, we're just setting vars.  could we just make that as part of a `prepare()` function?

File: run.sh [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+# Helper for getting absolute paths.
+function absPath() {
+  (
+  cd "$(dirname $1)"
+  echo "$PWD/$(basename $1)"
+  )
+}
+
+main "$@"
```
Comment (fridiculous) [inline run.sh:89]: How would someone know which arguments to send?   Should we put that in a readme?

Although i'm realizing that this script takes either no arguments.  Would it make sense to make it invoke a usage method if empty instead?

```
usage() {
  echo "USAGE: ${BASH_SOURCE[0]} <.> ..."
  echo ""
  echo "Example:
  ${BASH_SOURCE[0]} <a> ....
  ${BASH_SOURCE[0]} <b> ....
  "
  exit 0
}
```

**Pull request merged**